draft-ietf-ppm-dap.md

```diff
@@ -1377,20 +1389,30 @@ This message consists of:
 * `prepare_inits`: the sequence of `PrepareInit` messages constructed in the
   previous step.
 
-Finally, the Leader sends a PUT request to
-`{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}`. The payload
-is set to `AggregationJobInitReq` and the media type is set to
-"application/dap-aggregation-job-init-req".
+Finally, the Leader sends an HTTP PUT request to
+`{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}` with a media
+type of "application/dap-aggregation-job-init-req" and a body containing the
+`AggregationJobInitReq`.
 
 The Leader MUST authenticate its requests to the Helper using a scheme that
 meets the requirements in {{request-authentication}}.
 
-The Helper's response will be an `AggregationJobResp` message (see
-{{aggregation-helper-init}}. The response's `prepare_resps` must include exactly
-the same report IDs in the same order as the Leader's `AggregationJobInitReq`.
-Otherwise, the Leader MUST abort the aggregation job.
-
-[[OPEN ISSUE: consider relaxing this ordering constraint. See issue#217.]]
+The Helper responds with HTTP status 201 Created. If the response has a media
```

Comment:
Shouldn't this response include a `Location` field indicating what was created?

Reply:
I assume you mean https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location?

Great question. We actually specify the path for the subsequent GET in this PR, so it might not be strictly necessary?

One potential oddness here: If the helper responds synchronously, i.e., the body of the response contains the `AggregationJobResp`, so there would be no subsequent GET, and therefore no Location to return. Perhaps the leader should respond with 200 OK in this case?

Reply:
I think no action is required here. Per https://www.rfc-editor.org/rfc/rfc9110#section-15.3.2-1

> The primary resource created by the request is identified by either a [Location](https://www.rfc-editor.org/rfc/rfc9110#field.location) header field in the response or, if no [Location](https://www.rfc-editor.org/rfc/rfc9110#field.location) header field is received, by the target URI.

since the PUT and GET URIs are the same, we shouldn't need a `Location` header. (please correct me if I'm misreading)

Reply:
Ah, my mistake, the PUT/POST and GET URIs are not actually the same since we don't require the step field in the PUT/POST request--in which case yes we should have a Location header.